src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -121,9 +120,53 @@ internal void CopyModelConditionFlags(BitArray<ModelConditionFlag> newFlags)
 
         public readonly ObjectDefinition Definition;
 
-        public readonly Transform Transform;
+        private readonly Transform _transform;
         public readonly Transform ModelTransform;
 
+        private void UpdateCollider()
+        {
+            Collider.Update(_transform);
+            _gameContext.Quadtree.Update(this);
+        }
+
+        public Vector3 EulerAngles => _transform.EulerAngles;
+        public Vector3 LookDirection => _transform.LookDirection;
+        public Vector3 Translation => _transform.Translation;
+        public Quaternion Rotation => _transform.Rotation;
+        public Matrix4x4 TransformMatrix => _transform.Matrix;
```

Comment:
Does this copy the matrix? Could it return a `ref readonly` instead?

Reply:
ref readonly does not seem to work here